functions.php

```diff
@@ -3223,5 +3223,78 @@ function add_classic_degree_body_class( $classes ) {
 
 add_action( 'body_class', 'add_classic_degree_body_class' );
 
+function add_colleges_fields( $taxonomy ) {
+	wp_enqueue_script('media-upload');
+?>
+	<div class="form-field term-group">
+		<label for="colleges_header_image">Header Image</label>
+		<button type="button" class="button" id="colleges_header_image_upload">Upload Image</button>
+		<input type="hidden" name="colleges_header_image" id="colleges_header_image">
+	</div>
+	<div class="form-field term-group">
+		<img id="colleges_header_image_preview" src="" style="width: 100%;">
+	</div>
+<?php
+}
+
+add_action( 'colleges_add_form_fields', 'add_colleges_fields', 10, 1 );
+
+function edit_colleges_fields( $term, $taxonomy ) {
+	wp_enqueue_script('media-upload');
+	$image = get_term_meta( $term->term_id, 'colleges_header_image', true );
+?>
+	<tr class="form-field term-group">
+		<th scope="row"><label for="colleges_header_image">Header Image</label></th>
+		<td>
+			<button type="button" class="button" id="colleges_header_image_upload">Change Image</button>
+			<button type="button" class="button" id="colleges_header_image_remove">Remove Image</button>
+		</td>
+		<td><input type="hidden" name="colleges_header_image" id="colleges_header_image"></td>
+	</tr>
+	<tr class="form-field term-group">
+		<td>Preview: </td>
+		<td><img id="colleges_header_image_preview" src="<?php echo $image; ?>" style="width: 100%;" ></td>
+	</tr>
+<?php
+}
+
+add_action( 'colleges_edit_form_fields', 'edit_colleges_fields', 10, 2 );
+
+function save_colleges_meta( $term_id, $tt_id ) {
+	if ( isset( $_POST['colleges_header_image'] ) ) {
+		$header_image = $_POST['colleges_header_image'];
+		add_term_meta( $term_id, 'colleges_header_image', $header_image );
+	}
+}
+
+add_action( 'created_colleges', save_colleges_meta, 10, 2 );
+
+function edit_colleges_meta( $term_id, $tt_id ) {
+	if ( isset( $_POST['colleges_header_image'] ) ) {
+		$header_image = $_POST['colleges_header_image'];
+		update_term_meta( $term_id, 'colleges_header_image', $header_image );
+	}
+}
+
+add_action( 'edited_colleges', edit_colleges_meta, 10, 2 );
+
+/**
+ * Get Degree Header Image logic
+ **/
+function get_degree_header_image() {
+	global $post, $theme_options;
```

Comment:
Just using get_theme_option('theme_option_name') to fetch theme options is preferred instead of referencing the global $theme_options object